### PR TITLE
Template 1.14 update

### PR DESCRIPTION
**Pull request by ewels**
Attempting to bring in the template updates from nf-core/tools version 1.14.

Addresses https://github.com/nf-core/dualrnaseq/issues/14

Needs some careful checking! Done a first pass here but need to sanity check that I haven't made any mistakes..

### CODE REVIEW

**Comment by ewels** [comment]
Linting failing, needs some more work. Will come back to this shortly.

**Comment by ewels** [comment]
Note that although it's fine to use arrays in nextflow configs and JSON schema, we (currently) don't allow this because (a) it can't easily be supplied as a command line argument to the pipeline and (b) none of the nf-core tooling supports it yet for the schema / launch commands etc.

I have converted the lists to csv strings, stealing the conversion back again from [this Sarek code](https://github.com/nf-core/sarek/blob/68b9930a74962f3c42eee71f51e6dd2646269199/main.nf#L68).

Need to check that it still actually works.. 😅

**Comment by ewels** [comment]
Ok, nearly there - now it's just my list -> string conversion thing that looks buggy:

```bash
python /home/runner/work/dualrnaseq/dualrnaseq/bin/extract_annotations_from_gff.py \
  -gff SL1344_sub_parent_attribute.gff3 \
  -f [gene,sRNA,tRNA,rRNA] \
  -a parent -org pathogen -q_tool salmon -o pathogen_gff_annotations
```

Need to go back to the original code and figure out what this command should look like..

File: nextflow.config [hunk 1]
```diff
@@ -169,14 +171,6 @@ params {
 }
 
 
-//--------
-// Work directory:
-//--------
-//This location stores the working temporary files and downloaded images
-//Note: If commented out - errors sometimes occur re failing to publish files (although files are published correctly)
-workDir = "$projectDir/work"
```
Comment (ewels) [inline nextflow.config:177]: It's a bad idea to set this config option like this. The standard way of running is to do:

```bash
nextflow run nf-core/dualrnaseq ...
```

Which will fetch the pipeline files from GitHub automatically and save them to `$NXF_HOME`. By default, this in `~/.nextflow`.

With the setting above, all intermediate pipeline files will be stored inside the pipeline source directory here. On many HPC systems, the user's home directory has a very small storage allocation and all analysis work should be done in project folders. So by default, this setting will break on those systems.

If this was added due to problems, the affected users should set work dir manually. It shouldn't be set in the pipeline code and it definitely shouldn't be inside the pipeline source code directory 😉

File: nextflow.config [hunk 2]
```diff
@@ -186,61 +180,79 @@ process.container = 'nfcore/dualrnaseq:dev'
...
-//} catch (Exception e) {
-//  System.err.println("WARNING: Could not load nf-core/config/dualrnaseq profiles: ${params.custom_config_base}/pipeline/dualrnaseq.config")
-//}
-
-// to read your own custom genome file named computing_platform.config and saved in conf folder. 
-//includeConfig 'conf/computing_platform.config'                                                             
-
-
-// Load genomes.config if required 
-if (!params.genomes_ignore) {
```
Comment (ewels) [inline nextflow.config:220]: Duplicate of above

File: nextflow.config [hunk 3]
```diff
@@ -186,61 +180,79 @@ process.container = 'nfcore/dualrnaseq:dev'
...
 
-// Load nf-core/dualrnaseq custom profiles from different Institutions
-//try {
-//  includeConfig "${params.custom_config_base}/pipeline/dualrnaseq.config"
-//} catch (Exception e) {
-//  System.err.println("WARNING: Could not load nf-core/config/dualrnaseq profiles: ${params.custom_config_base}/pipeline/dualrnaseq.config")
-//}
-
-// to read your own custom genome file named computing_platform.config and saved in conf folder. 
-//includeConfig 'conf/computing_platform.config'
```
Comment (ewels) [inline nextflow.config:216]: This is a something that I will be harping on about in my upcoming PRs too.

See https://nf-co.re/usage/configuration#different-config-locations :

> ### You do not need to edit the pipeline code to configure nf-core pipelines!
> If you edit the pipeline defaults then you cannot update to more recent versions of the pipeline without overwriting your changes. You also run the risk of moving away from the canonical pipeline and losing reproducibility.

In this case, if people want to write their own custom config files, they should do so in one of these locations: https://nf-co.re/usage/configuration#custom-configuration-files

File: nextflow.config (hunk 1 above)
Comment (reganhayward) [reply nextflow.config:177]: I agree, but we had some errors during runtime which said files failed to be published - although the files were published correctly. This seemed to fix the issue.

Comment (ewels) [reply nextflow.config:177]: Yeah, I figured from the comment. Can you remember the details? We just need to fix in a different way I think..

**Review by reganhayward — APPROVED** [review]

**Pull request merged**